lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp

```diff
@@ -978,20 +980,27 @@ int amf_app_handle_pdu_session_accept(
   -------------------------------------- */
   smf_msg->msg.pdu_session_estab_accept.nssai.iei =
       static_cast<uint8_t>(M5GIei::S_NSSAI);
-  if (smf_ctx->requested_nssai.sst) {
-    if (smf_ctx->requested_nssai.sd[0]) {
-      smf_msg->msg.pdu_session_estab_accept.nssai.len = SST_LENGTH + SD_LENGTH;
-      smf_msg->msg.pdu_session_estab_accept.nssai.sst =
-          smf_ctx->requested_nssai.sst;
-      memcpy(smf_msg->msg.pdu_session_estab_accept.nssai.sd,
-             smf_ctx->requested_nssai.sd, 3);
-    } else {
-      smf_msg->msg.pdu_session_estab_accept.nssai.len = SST_LENGTH;
-      smf_msg->msg.pdu_session_estab_accept.nssai.sst =
-          smf_ctx->requested_nssai.sst;
-    }
-    buf_len = smf_msg->msg.pdu_session_estab_accept.nssai.len + 2;
+
+  s_nssai_t slice_information = {};
+  amf_smf_get_slice_configuration(smf_ctx, &(slice_information));
+  if (!slice_information.sst) {
+    OAILOG_ERROR(LOG_AMF_APP,
+                 "Slice Configuration does not exist:" AMF_UE_NGAP_ID_FMT,
+                 ue_id);
+
+    return M5G_AS_FAILURE;
+  }
+
+  if (slice_information.sd[0]) {
+    smf_msg->msg.pdu_session_estab_accept.nssai.len = SST_LENGTH + SD_LENGTH;
+    smf_msg->msg.pdu_session_estab_accept.nssai.sst = slice_information.sst;
+    memcpy(smf_msg->msg.pdu_session_estab_accept.nssai.sd, slice_information.sd,
+           3);
```

Comment:
Can we use SD_LENGTH instead of 3.

Reply:
Comment addressed.

Reply:
Comment Addressed.

Reply:
I modified it to SD_LENGTH in place of 3, thanks.